bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java

```diff
@@ -342,6 +354,9 @@ public boolean check(State input) {
                     if (rhsFunction.alwaysAccept()) {
                         return true;
                     }
+                    if (rhsFunction.getType() == FunctionType.Function.DELTA) {
+                        isDeltaCheck = true;
+                    }
```

Comment:
This does seems like a better place for it, so can we remove the same code further down, i.e. line 415-418?

Reply:
> can we remove the same code further down

Umm not sure. the first sets `isDeltaCheck` if LHS is a function, and the second does it if RHS is. ??

Reply:
I'm not talking about the lhs (line 404-406). I'm talking about line 415-418 - that's also rhs.

Reply:
Ah yes.